src/main/java/jp/kusumotolab/kgenprog/ga/LazyVariantFactory.java

```diff
@@ -0,0 +1,38 @@
+package jp.kusumotolab.kgenprog.ga;
+
+import java.util.List;
+import io.reactivex.Single;
+import io.reactivex.schedulers.Schedulers;
+import jp.kusumotolab.kgenprog.Counter;
+import jp.kusumotolab.kgenprog.OrdinalNumber;
+import jp.kusumotolab.kgenprog.Strategies;
+import jp.kusumotolab.kgenprog.fl.Suspiciousness;
+import jp.kusumotolab.kgenprog.project.GeneratedSourceCode;
+import jp.kusumotolab.kgenprog.project.test.TestResults;
+
+public class LazyVariantFactory implements VariantFactory {
+
+  @Override
+  public Variant exec(final OrdinalNumber generation, final Counter variantCounter, final Gene gene,
+      final GeneratedSourceCode sourceCode, final HistoricalElement element,
+      final Strategies strategies) {
+
+    final Single<TestResults> resultsSingle = Single.create(
+        emitter -> emitter.onSuccess(strategies.execTestExecutor(sourceCode)))
```

Comment:
```suggestion
       (SingleEmitter<TestResults> emitter) -> emitter.onSuccess(strategies.execTestExecutor(sourceCode)))
```

ここで型を明示すると、23行目のcastをなくせます